frame/staking-rewards/src/lib.rs

```diff
@@ -372,6 +388,61 @@ pub mod pallet {
 			<Self as Staking>::split(&who, &position, ratio.value())?;
 			Ok(())
 		}
+
+		/// Update a new reward pool based on the provided amounts and rates.
+		///
+		/// Emits `RewardPoolUpdated` event when successful.
+		#[pallet::weight(T::WeightInfo::update_reward_pool())]
+		#[transactional]
+		pub fn update_reward_pool(
+			origin: OriginFor<T>,
+			pool_id: T::RewardPoolId,
+			reward_updates: RewardUpdatesOf<T>,
+		) -> DispatchResult {
+			T::RewardPoolUpdateOrigin::ensure_origin(origin)?;
+
+			let current_time = T::UnixTime::now().as_secs();
+
+			let mut rewards_pool =
+				RewardPools::<T>::try_get(pool_id).map_err(|_| Error::<T>::RewardsPoolNotFound)?;
+
+			let mut inner_rewards = rewards_pool.rewards.into_inner();
+
+			let elapsed_time = current_time - rewards_pool.last_updated;
```

Comment:
I was going to suggest safe_sub here but `elapsed_time` appears to be unused?

Reply:
it is unused right now until we get a common solution on how to handle `elapsed_time * reward_rate`, but `safe_sub` makes sense